lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -746,6 +746,98 @@ class PropagateAtenItemPattern : public OpRewritePattern<AtenItemOp> {
 };
 } // namespace
 
+namespace {
+
+LogicalResult convertOpFoldResults(ImplicitLocOpBuilder &b,
+                                   SmallVector<OpFoldResult> &converted,
+                                   SmallVector<OpFoldResult> &elements,
+                                   Type inputDtype, Type resultDtype) {
+  auto inputIsInt = dyn_cast<mlir::IntegerType>(inputDtype);
```

Comment:
Probably should cache similarly for `fp` so the checks below just check each global.

Reply:
I think I'm going to skip this and keep the inlined `isa` checks.